tdrs-backend/tdpservice/reports/validators.py

```diff
@@ -0,0 +1,88 @@
+"""Validator functions which may be assigned to model fields."""
+import logging
+import re
+
+from django.core.exceptions import ValidationError
+from inflection import pluralize
+
+from tdpservice.clients import ClamAVClient
+
+logger = logging.getLogger(__name__)
+
+# The default set of Extensions allowed for an uploaded file
+# Supports regex patterns as defined by standard library
+# https://docs.python.org/3/library/re.html#regular-expression-syntax
+ALLOWED_FILE_EXTENSIONS = [
+    r'^(ms\d{2})$',  # Files ending in .MS## where # is a digit 0-9
+    r'^(ts\d{2,3})$',  # Files ending in .TS## or .TS### where # is a digit 0-9
+    'txt',  # plain text files
+]
+
+# The default set of Content Types allowed for an uploaded file
+ALLOWED_FILE_CONTENT_TYPES = [
+    'text/plain',
+]
+
+
+def _get_unsupported_msg(_type, value, supported_options):
+    """Construct a message to convey an unsupported operation."""
+    return (
+        f'Unsupported {_type}: {value}, supported {pluralize(_type)} '
+        f'are: {supported_options}'
+    )
+
+
+def validate_file_content_type(file):
+    """Validate the content type of a file is in our supported list."""
+    file_content_type = file.content_type
+
+    if file_content_type not in ALLOWED_FILE_CONTENT_TYPES:
+        msg = _get_unsupported_msg(
+            'file content type',
+            file_content_type,
+            ALLOWED_FILE_CONTENT_TYPES
+        )
+        raise ValidationError(msg)
+
+
+def validate_file_extension(file_name: str):
+    """Validate the file extension of a file is in our supported list."""
+    file_extension = (
+        file_name.split('.')[-1].lower() if '.' in file_name else None
+    )
+
+    allowed_ext_patterns = '|'.join(ALLOWED_FILE_EXTENSIONS)
+    if (
+        file_extension is not None
+        and not re.match(allowed_ext_patterns, file_extension)
+    ):
```

Comment:
This uses regular expressions to match patterns defined above so we don't need to keep an exhaustive list of all possible file endings but can still assert that files matching patterns like `ADS.E2J.FTP1.TS06` are still accepted.

Reply:
perfect solution! thank you @jtwillis92 !